### ISSUE TITLE
Intellisense-style autocomplete doesn't work for zustand/middleware

### ISSUE THREAD

**Issue by dai-shi** [pmndrs/zustand#262]
from: https://github.com/pmndrs/zustand/pull/248#issuecomment-740303328

This is an issue to track the titled issue.
(I hope this will be resolved when TypeScript supports "exports" in package.json.)

**Comment by dai-shi** [comment]
Closing this as this is not actionable now. Let's re-open it when things change.
If anyone is interested, feel free to investigate it.

**Comment by TurtIeSocks** [comment]
@dai-shi apologies for bumping a really old issue but I've noticed a small bug that _might_ be related and wasn't sure if I should open a new issue or if I'm doing something incorrectly.

I have an older project I wrote in JS and is just a little too tedious to port entirely to TS, and I noticed that the intellisense doesn't seem to work when `set` and `get` (or only `set`) arguments are used when creating the store.

For example:
Creating this store:
![Screenshot 2023-07-21 at 2 38 01 PM](https://github.com/pmndrs/zustand/assets/58572875/730825b1-5414-4427-8fa4-f4f6312c4c60)
Results in this intellisense:
![Screenshot 2023-07-21 at 2 37 49 PM](https://github.com/pmndrs/zustand/assets/58572875/64f39851-8763-4875-bf4d-59156b8ec88b)

But creating this store:
![Screenshot 2023-07-21 at 2 38 12 PM](https://github.com/pmndrs/zustand/assets/58572875/10ed1d13-eeec-4688-aa1c-71ef1d383b8d)
Results in no intellisense:
![Screenshot 2023-07-21 at 2 38 27 PM](https://github.com/pmndrs/zustand/assets/58572875/8199c41f-105f-43c4-8725-1ad2edb620c8)

The results are the same for each scenario when wrapping the store with the `persist` middleware.

Version:
`"zustand": "^4.3.9"`

Zustand is imported like so (if it's helpful to know)
```js
import { create } from 'zustand'
import { persist } from 'zustand/middleware'
```

Any suggestions? 

Thank you for this wonderful lib! 🙏
